styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -553,28 +567,34 @@ private void setupMetrics(
       WorkflowCache cache,
       WorkflowInitializer workflowInitializer,
       Stats stats,
-      StateManager stateManager) {
-
+      StateManager stateManager,
+      BiConsumer<Optional<Workflow>, Optional<Workflow>> workflowConsumer) {
     return (workflow) -> {
       stats.registerActiveStatesMetric(
           workflow.id(),
           () -> stateManager.getActiveStatesCount(workflow.id()));
 
-      cache.store(workflow);
+      final Optional<Workflow> oldWorkflowOptional = cache.workflow(workflow.id());
+
       workflowInitializer.inspectChange(workflow);
+      cache.store(workflow);
```

Comment:
This slightly changed the behaviour. if `inspectChange` fails, we will not update cache, which makes more sense.

Reply:
The change was intentional ;)